src/cljc/matcher_combinators/matchers.cljc

```diff
@@ -31,6 +31,19 @@
         :else
         (core/->Value expected)))
 
+(declare match-with)
+
+(defn deep-equals
+  "Return a matcher for the expected value which matches nested maps with `equals`.
+  This solves a common need of matching nested maps with strict equality.
+  See also: `match-with`.
```

Comment:
Feel free to suggest a better explanation here

Reply:
Maybe something more like the following which explains it in broader terms and mentions the fact that there are defaults that are being overridden:
> A matcher that always uses the `equals` matcher at every level of nesting. Useful given that matchers usually only change the first level of the data they are applied to, leaving nested data to use the default matcher of that type of data. For instance, this can be used to assert that any nested map has exactly the same keys and matching values as provided in the `expected`, and no more.
>
> Note: this excludes functions, which continue to be invoked as predicates instead of compared via the `equals` matcher

Reply:
I have added `regex` exclusion too. Feel free to correct me if I'm wrong